install.sh

```diff
@@ -1,32 +1,64 @@
-#!/bin/bash
-
-get_os () {
-  if [[ "$OSTYPE" == "linux-gnu" ]]; then
-    echo "linux"
-  elif [[ "$OSTYPE" == "freebsd"* ]]; then
-    echo "linux"
-  elif [[ "$OSTYPE" == "darwin"* ]]; then
-    echo "macos"
-  elif [[ "$OSTYPE" == "cygwin" ]]; then
-    echo "windows"
-  elif [[ "$OSTYPE" == "msys" ]]; then
-    echo "windows"
-  elif [[ "$OSTYPE" == "win32" ]]; then
-    echo "windows"
+#!/bin/sh
+
+set -eu
+
+DESTDIR="${DESTDIR:-/}"
+PREFIX="${PREFIX:-"$DESTDIR/usr/local"}"
+
+releases_url="https://github.com/soywod/himalaya/releases"
+
+uname_os() {
+  os=$(uname -s | tr '[:upper:]' '[:lower:]')
+
+  case $os in
+    msys*) os="windows" ;;
+    mingw*) os="windows" ;;
+    cygwin*) os="windows" ;;
+    win*) os="windows" ;;
+  esac
+
+  echo "$os"
+}
+
+get_system() {
+  case $(uname_os) in
+    linux | freebsd) system="linux" ;;
+    darwin) system="macos" ;;
+    windows) system="windows" ;;
```

Comment:
`uname_os` and `get_system` cannot be merged together?

```suggestion
    msys*) os="windows" ;;
    mingw*) os="windows" ;;
    cygwin*) os="windows" ;;
    win*) os="windows" ;;
```